### PR TITLE
drop legacy arguments encoding with named tuple codec

### PR DESCRIPTION
**Pull request by nsidnev**
Continues https://github.com/edgedb/edgedb-python/pull/311

Though I haven't received an answer as to whether it's time to drop the legacy encoding, so if it's too early, feel free to close this PR.

### CODE REVIEW

File: edgedb/protocol/protocol.pyx [hunk 1]
```diff
@@ -1089,22 +1089,15 @@ cdef class SansIOProtocol:
 
         in_dc_type = type(in_dc)
 
-        if in_dc_type in {NullCodec, EmptyTupleCodec}:
-            # TODO: drop EmptyTupleCodec when 1.0 RC1 is released.
```
Comment (nsidnev) [inline edgedb/protocol/protocol.pyx:1093]: I am not quite sure if that means removing the codec only from this condition or removing the codec completely

**Comment by nsidnev** [comment]
@fantix friendly ping :)

File: edgedb/protocol/protocol.pyx (hunk 1 above)
Comment (fantix) [reply edgedb/protocol/protocol.pyx:1089]: We can address dropping `EmptyTupleCodec` itself in a separate PR - it shouldn't be an issue keeping it I think.

**Review by fantix — APPROVED** [review]
Thanks for the PR and sorry for the delay! There was some discussion and it eventually came to #318.

This PR looks good! I'll rebase #318 on top of this.

**Pull request merged**